src/common/scratchpad.cpp

```diff
@@ -74,14 +74,14 @@ struct global_scratchpad_t : public scratchpad_t {
     virtual char *get() const { return scratchpad_; }
 
 private:
-    thread_local static char *scratchpad_;
-    thread_local static size_t size_;
-    thread_local static unsigned int reference_count_;
+    static char *scratchpad_;
+    static size_t size_;
+    static unsigned int reference_count_;
 };
 
-thread_local char *global_scratchpad_t::scratchpad_ = nullptr;
-thread_local size_t global_scratchpad_t::size_ = 0;
-thread_local unsigned int global_scratchpad_t::reference_count_ = 0;
+char *global_scratchpad_t::scratchpad_ = nullptr;
+size_t global_scratchpad_t::size_ = 0;
+unsigned int global_scratchpad_t::reference_count_ = 0;
```

Comment:
This sounds to me as a severe library problem, which should have a separate test somewhere in `tests/other".